service/service.go

```diff
@@ -0,0 +1,46 @@
+package service
+
+import (
+	"fmt"
+	"github.com/majustman/utility-tree/tree"
+)
+
+func Run(address string, filesParam bool) error {
+	t, err := tree.NewTree(address)
+	if err != nil {
+		return err
+	}
+	printTree(t, filesParam, 0)
+	return nil
+}
+
+func printTree(t *tree.Tree, fileParam bool, level int) {
+	if level == 0 {	fmt.Println(t.Name)	}
+	if fileParam { printFiles(t.Files, level, len(t.Dirs)) }
+	for i, d := range t.Dirs {
+		printSpace(level)
+		if i != len(t.Dirs)-1 {
+			fmt.Printf("├── %v\n", d.Name)
```

Comment:
In my opinion, in this type of scenario, when you print many formatted outputs it's suboptimal to use `fmt` functions, because it adds overhead. You might want to write your own simple formatter with `strings.Builder` for this case. This will result in performance improvement.

The following article covers the basics, but there are many other articles on this topic.
e.g. article: https://yourbasic.org/golang/build-append-concatenate-strings-efficiently/

Reply:
Implemented strings.Builder